eng/CodeGeneration.targets

```diff
@@ -5,18 +5,20 @@
     <_AutoRestCoreVersion>3.0.6257</_AutoRestCoreVersion>
     <_AutoRestCSharpVersion>https://github.com/Azure/autorest.csharp/releases/download/3.0.0-dev.20200323.2/autorest-csharp-v3-3.0.0-dev.20200323.2.tgz</_AutoRestCSharpVersion>
     <_SupportsCodeGeneration Condition="'$(IsClientLibrary)' == 'true'">true</_SupportsCodeGeneration>
-    <AutoRestConfiguration Condition="'$(AutoRestConfiguration)' == ''">autorest.md</AutoRestConfiguration>
-
+    <_DefaultInputName>$(MSBuildProjectDirectory)/autorest.md</_DefaultInputName>
+    <AutorestInput Condition="'$(AutorestInput)' == '' AND Exists('$(_DefaultInputName)')">$(_DefaultInputName)</AutorestInput>
+    <AutorestConfiguration Condition="'$(AutorestConfiguration)' == '' AND '$(AutorestInput)' != '$(_DefaultInputName)'">$(_DefaultInputName)</AutorestConfiguration>
+    <AutorestAdditionalParameters></AutorestAdditionalParameters>
     <_SharedCodeDirectory>$(MSBuildThisFileDirectory)../sdk/core/Azure.Core/src/Shared/</_SharedCodeDirectory>
     <_AutoRestSharedCodeDirectory>$(_SharedCodeDirectory)Autorest/</_AutoRestSharedCodeDirectory>
 
-    <_GenerateCode Condition="'$(_SupportsCodeGeneration)' == 'true' AND Exists('$(AutoRestConfiguration)')">true</_GenerateCode>
+    <_GenerateCode Condition="'$(_SupportsCodeGeneration)' == 'true' AND '$(AutorestInput)' != ''">true</_GenerateCode>
     <UsesJsonSerialization Condition="'$(UsesJsonSerialization)' == ''">true</UsesJsonSerialization>
   </PropertyGroup>
 
   <Target Name="GenerateCode" Condition="'$(_GenerateCode)' == 'true'" >
     <RemoveDir Directories="$(MSBuildProjectDirectory)/Generated"/>
-    <Exec Command="npx autorest@$(_AutoRestVersion) --version=$(_AutoRestCoreVersion) --input=$(AutoRestConfiguration) --use=$(_AutoRestCSharpVersion) --output-folder=$(MSBuildProjectDirectory) --title=$(RootNamespace) --namespace=$(RootNamespace) --shared-source-folder=$(_SharedCodeDirectory) --verbose" />
+    <Exec Command="npx autorest@$(_AutoRestVersion) --version=$(_AutoRestCoreVersion) $(AutorestInput) --require=$(AutorestConfiguration) $(AutorestAdditionalParameters) --use=$(_AutoRestCSharpVersion) --output-folder=$(MSBuildProjectDirectory) --title=$(RootNamespace) --namespace=$(RootNamespace) --shared-source-folder=$(_SharedCodeDirectory) --verbose" />
```

Comment:
normally the namespace is specified in central configuration as well, e.g.
https://github.com/Azure/azure-rest-api-specs/blob/master/specification/storage/resource-manager/readme.csharp.md

Reply:
And it's wrong there for track 2 libraries.

Reply:
We are working on figuring out configuration for track 2 libraries.

Reply:
Added one issue https://github.com/Azure/autorest.csharp/issues/604 for track purpose